src/ScType.ts

```diff
@@ -121,12 +174,94 @@ export class ScType {
     return this._value === other._value;
   }
 
-  public merge(other: ScType): ScType {
-    const t1 = this._value & sc_type_element_mask;
-    const t2 = other._value & sc_type_element_mask;
+  protected isNotCompatibleByMask(newType: ScType, mask: number): boolean {
+    const subtype = this.value & mask;
+    const newSubtype = newType.value & mask;
+    return subtype != sc_type_unknown && subtype != newSubtype;
+  }
```

Comment:
this method can be static that receives two ints and mask, no need to construct new ScType just to pass it here and get its value

Reply:
what's the advantage?

Reply:
same as in py-sc-client

Reply:
fixed